File: client/src/app/shared/Course.ts
import { IStudentCourse } from "./StudentCourse";
import { ISubject } from "./Subject";

export class ICourse {
    courseId: number;
    courseName: string;
    title: string;
    studentCourses: IStudentCourse[];
    subjects: ISubject[];
    totalTeachers: number;
    totalStudents: number;
    averageOfGrades: number;
}

export class ICreateCourse {
    courseName: string;
    title: string;
}

File: Data/Entities/Course.cs
using System.Collections.Generic;

namespace Magnifinance.Data.Entities
{
    public class Course
    {
        public int CourseId { get; set; }
        public string CourseName { get; set; }
        public string Title { get; set; }

        //public List<TeacherCourse> TeacherCourses { get; set; }
        public List<StudentCourse> StudentCourses { get; set; }

        public List<Subject> Subjects { get; set; }
    }
}
